server/service/users.go

```diff
@@ -574,6 +574,12 @@ func (svc *Service) modifyEmailAddress(ctx context.Context, user *fleet.User, em
 		return err
 	}
 	token := base64.URLEncoding.EncodeToString([]byte(random))
+
+	_, err = svc.ds.UserByEmail(ctx, email)
+	if err == nil {
+		return ctxerr.Wrap(ctx, alreadyExistsError{})
+	}
```

Comment:
Here you are assuming `err` == `sql.ErrNoRows`, to be on safe side you could do something like:
```go
switch _, err = svc.ds.UserByEmail(ctx, email); {
case err == nil:
  return ctxerr.Wrap(ctx, alreadyExistsError{})
case errors.Is(err, sql.ErrNoRows):
  // OK
default:
  return ctxerr.Wrap(ctx, err)
}
...
```